### PR TITLE
feat: support projectset logicalPlan

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: internal/topo/planner/planner.go [hunk 1]
```diff
@@ -371,19 +375,70 @@ func createLogicalPlan(stmt *ast.SelectStatement, opt *api.RuleOption, store kv.
...
+	})
+	return err
+}
+
+// extractSRFFromFileds extracts the set-returning-functions in the select field.
+func extractSRFFromFileds(stmt *ast.SelectStatement) (map[string]struct{}, error) {
+	srfCount := 0
+	nonSrfCount := 0
+	srfMapping := make(map[string]struct{}, 0)
+	for _, field := range stmt.Fields {
```
Comment (ngjaying) [inline internal/topo/planner/planner.go:425]: how about nest unnest function, like abs(unnest(a))

File: internal/topo/planner/ProjectSetPlan.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package planner
+
+type ProjectSetPlan struct {
```
Comment (ngjaying) [inline internal/topo/planner/ProjectSetPlan.go:17]: I am not sure if it is necessary to have project & projectSet together in one plan. Can you find the benefits when doing implementation?

Comment (Yisaer) [reply internal/topo/planner/ProjectSetPlan.go:17]: We will use `ProjectOp` for `ProjectSet` Executor. Thus we directly build a built-in `ProjectPlan` for `ProjectSetPlan` in order to build `ProjectOp` by `ProjectPlan`.

On the other hand, `ProjectSetPlan` do quite the similar thing as `ProjectPlan`, except `ProjectSet` will handle the `srf`

File: internal/topo/planner/planner.go (hunk 1 above)
Comment (Yisaer) [reply internal/topo/planner/planner.go:425]: good catch, we should validate that `srf` shouldn't be nested for now.

File: internal/topo/planner/ProjectSetPlan.go (hunk 2 above)
Comment (ngjaying) [reply internal/topo/planner/ProjectSetPlan.go:17]: I have no idea now. Just think about if it is better to keep each plan "single responsibility". We can pipe from SRF to project.

Comment (Yisaer) [reply internal/topo/planner/ProjectSetPlan.go:17]: I got it. So the final plan will be `ProjectSet` -> `Project` -> .... , that's ok to me.

Comment (ngjaying) [reply internal/topo/planner/ProjectSetPlan.go:17]: Just an alternative way. During implementation, you may have more ideas about which way is better.

**Pull request closed without merging**